### PR TITLE
Improve accuracy of swagger output

### PR DESCRIPTION
**Pull request by rygwdn**
- Allow omitting endpoints on the viewset
- Improve default status codes
- DELETE responses are empty
- add hooks to augment operations
- add support for pagination
- remove trailing "Serializer" from definitions

### CODE REVIEW

File: rest_framework_swagger/docgenerator.py [hunk 1]
```diff
@@ -139,29 +151,68 @@ def get_operations(self, api_endpoint, introspector):
...
-            response_messages['200'] = {
-                'description': 'Successful operation',
-                'schema': {
-                    '$ref': '#/definitions/' + response_type
-                } if response_type != 'object' else {
-                    'type': response_type
-                }
-            }
+            # Remove blank response objects - allows yaml to remove default responses
+            for code in response_messages:
```
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: Sketchy modification of collection while iterating on it, could explicitly make a copy of keys before iterating or do a comprehension and create a new dict

File: rest_framework_swagger/docgenerator.py [hunk 2]
```diff
@@ -139,29 +151,68 @@ def get_operations(self, api_endpoint, introspector):
...
+                filter_(operation, callback=method_introspector.callback, method=method_introspector.method)
 
             operations.append(operation)
 
         return operations
 
+    def _get_default_response_object(self, operation_method, response_type):
+        schema = {
+            '$ref': '#/definitions/' + response_type
+        } if response_type != 'object' else {
```
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: Not sold on this ternary being cleaner than an ordinary if/else

**Comment by npars** [comment]
:+1:

**Comment by mdentremont** [comment]
+2

**Pull request merged**